src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/HaFlowUpdateFsm.java

```diff
@@ -293,9 +296,14 @@ public Factory(@NonNull HaFlowGenericCarrier carrier, @NonNull Config config,
                     .perform(new HandleNotCompletedCommandsAction<>());
 
             builder.transition().from(State.NEW_RULES_REVERTED)
-                    .to(State.REVERTING_ALLOCATED_RESOURCES)
+                    .to(State.SKIP_REVERTING_RESOURCES_ALLOCATION)
                     .on(Event.NEXT);
 
+            builder.transitions().from(State.SKIP_REVERTING_RESOURCES_ALLOCATION)
+                    .toAmong(State.REVERTING_ALLOCATED_RESOURCES, State.REVERTING_FLOW)
+                    .onEach(Event.NEXT, Event.UPDATE_ENDPOINTS_ONLY)
+                    .perform(new SkipResourceManagementOnEndpointsUpdateAction(persistenceManager));
```

Comment:
I'm not sure it will work.

It say: 
go from state `SKIP_REVERTING_RESOURCES_ALLOCATION` to state `REVERTING_FLOW` on event `UPDATE_ENDPOINTS_ONLY`. After changing state please run `SkipResourceManagementOnEndpointsUpdateAction` 

But this action it the only action which can call `UPDATE_ENDPOINTS_ONLY` event. I think wemust call this action it the previous transition 
```
builder.transition().from(State.NEW_RULES_REVERTED)
                    .to(State.SKIP_REVERTING_RESOURCES_ALLOCATION)
                    .on(Event.NEXT);
```

Reply:
yes, I now think I was confused on how and when this framework manages transitions and events (thanks to their brilliant javadocs) I am double checking this now, But it is a bit tricky to reproduce and to test. Thanks for catching it!

Reply:
still, it is not the case that the action is executed after. I added tracing into history and it says that the action is executed while the state is what is specified in from clause. But the confusing point is when the event goes into action, as I understand now it takes effect in the next transition.